### DISCUSSION TITLE
grid patch sampler and mask

### DISCUSSION THREAD

**Discussion by UserA12345** [TorchIO-project/torchio#817] [Q&A]
Hey Everyone,
I am working on a project with brain mri. I would like to extract all the patches from the image for the training. First I use labelSampler and a brain_mask in order to avoid patches with no information. However as I need to extract all the patches from the image and not some random patches, following question #513 I used GridSampler rather than labelSampler but GridSampler couldn't handle a mask. Is there any way to use both please?
Thank you

**Comment by romainVala** [comment]
Hi
not sure to understand what you want to achieve with `GridSampler` handeling a mask ? (only patch within the mask ? then it is not a regular grid sampler any more ...)
Why not just `CropOrPad` the image (with the `mask_name` argument) to get rid of unnecessary background ?

**Reply by UserA12345** [reply]
thx, ideally I would have like an ordered sampling within the mask. I tried to used crop or pad but I need to provide a target_shape but as it is subject dependent I am not sure.

**Reply by romainVala** [reply]
I am still confuse about what will be an ordered sampling within the mask ... I can only see what is an ordered sampling within a grid (which is what GridSampler does)
what is subject dependent ? the input shape or the mask ?
if it is the mask, why not try do define a minimal box that will contains all you subject mask ?  Once you have it it will be used as target_shape in the `CropOrPad` transform ...

**Reply by UserA12345** [reply]
yes thx I will try that
